hermes-api/src/main/java/pl/allegro/tech/hermes/api/Subscription.java

```diff
@@ -42,59 +43,73 @@
     @Valid
     private BatchSubscriptionPolicy batchSubscriptionPolicy;
 
-    private boolean trackingEnabled;
+    private boolean trackingEnabled = false;
 
     @NotNull
     private String supportTeam;
 
     private String contact;
 
+    @NotNull
     private DeliveryType deliveryType = DeliveryType.SERIAL;
 
     public enum State {
         PENDING, ACTIVE, SUSPENDED
     }
 
-    private Subscription() {
-    }
-
-    public Subscription(TopicName topicName, String name, EndpointAddress endpoint, String description,
-                        Map<String, Object> subscriptionPolicy, boolean trackingEnabled, String supportTeam, String contact,
+    public Subscription(TopicName topicName,
+                        String name,
+                        EndpointAddress endpoint,
+                        State state,
+                        String description,
+                        Object subscriptionPolicy,
```

Comment:
I think that if this constructor is only accessed deliberately in test suite then it would be a good idea to have two versions with strong typing here.